worker/uniter/operation/runaction.go

```diff
@@ -73,6 +74,18 @@ func (ra *runAction) Execute(state State) (*State, error) {
 		return nil, err
 	}
 	err := ra.runner.RunAction(ra.name)
+	cause := errors.Cause(err)
+	switch {
+	case cause == context.ErrResetAndSleep:
+		err = ra.runner.Context().Reset()
+		if err != nil {
+			return nil, errors.Annotatef(err, "resetting action %q", ra.name)
+		}
+		return stateChange{
+			Kind: Continue,
+			Step: Pending,
```

Comment:
It seems with this approach, any event that wake sup the resolver will cause the action to be reset, rather than a pod status change. That's probably ok to start with.